### DISCUSSION TITLE
Where is `MeQuery` in `strawberry-django-auth`

### DISCUSSION THREAD

**Discussion by baseplate-admin** [nrbnlulu/strawberry-django-auth#410] [Q&A]
Hi there, Thanks for creating this library. 

So i have a question. I saw in documentation there's a [MeQuery](https://nrbnlulu.github.io/strawberry-django-auth/tutorial/#mequery). But i didn't see the code behind it.

Basically i want to see how to protect a query.

**Comment by nrbnlulu** [comment]
So it is here:

https://github.com/nrbnlulu/strawberry-django-auth/blob/d42e4ab0fa0dcc1d8c64a38e85451b4537ea8124/gqlauth/user/queries.py#L32

To protect a query you generally use permissions which you now create throw strawberry field extensions
or with the strawberry-django permissions (that are also field extensions.)

**Reply by baseplate-admin** [reply]
Hi, Could you add this to documentation?

Juggling between `strawberry-django` and `strawberry-django-auth` often scares new devs away :)

**Reply by nrbnlulu** [reply]
Sure

**Comment by baseplate-admin** [comment]
It seems that this code is not playing nice with `strawberry-django` `mutations`

So here's what i tried : 


```python
import strawberry
import strawberry_django
from strawberry_django.permissions import IsSuperuser

@strawberry.type
class Mutations:
    # Anime related things
    create_anime: Anime = strawberry_django.mutations.create(
        AnimeInput,
        # We allow only superuser to create data
        extensions=[
            IsSuperuser(),
        ],
    )
```


Screenshot : 

![image](https://github.com/nrbnlulu/strawberry-django-auth/assets/61817579/ef06bb90-e7fd-498b-8c68-2e6b53201c9d)

**Comment by baseplate-admin** [comment]
Following the approach in :  https://github.com/nrbnlulu/strawberry-django-auth/blob/d42e4ab0fa0dcc1d8c64a38e85451b4537ea8124/gqlauth/user/queries.py#L32-L37

Lead me to : 

```ts
{
  "data": null,
  "errors": [
    {
      "message": "'AnimeInput' object has no attribute 'context'",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "create_anime"
      ]
    }
  ]
}
```

With this code : 

```python

@strawberry_django.input(AnimeModel)
class AnimeInput:
    mal_id: auto
    anilist_id: auto
    kitsu_id: auto

    name: auto
    name_japanese: auto


@strawberry.mutation
def create_anime(self, data: AnimeInput) -> Anime:
    user = get_user(data)
    if not user.is_superuser:
        raise GQLAuthError(code=GQLAuthErrors.UNAUTHENTICATED)
    print(data)

```

**Reply by nrbnlulu** [reply]
you should pass `info: Info` there...  Read strawberry docs
